### PR TITLE
[SYCL][NFC] Refactor lit.cfg.py

### PR DESCRIPTION
**Pull request by AlexeySachkov**
Switched to use "lit_config.note" instead of just "print": this emits
potentially more usefult message, because it shows particular file and
line where it was emitted, something like:

```
llvm-lit: /sycl/test/lit.cfg.py:117: note: Found available CPU device
llvm-lit: /sycl/test/lit.cfg.py:154: warning: GPU device not found
llvm-lit: /sycl/test/lit.cfg.py:169: warning: Accelerator device not found
llvm-lit: /sycl/test/lit.cfg.py:198: warning: Couldn't find pre-installed AOT device compiler ocloc
llvm-lit: /sycl/test/lit.cfg.py:198: warning: Couldn't find pre-installed AOT device compiler aoc
llvm-lit: /sycl/test/lit.cfg.py:198: warning: Couldn't find pre-installed AOT device compiler opencl-aot
```

It also should be possible now to suppress this output via "-q" option
to llvm-lit

Removed extra items from "config.excludes" list since they were ignored
anyway based on content of "config.suffixes"

Simplified environment propagation by re-using existing helper functions

Updated logic of `getDeviceCount` function to avoid printing useless
messages

Removed duplicated handling of path to SYCL tools directory

Refactored handling of "opencl-aot" tool: now if it was built together
with product, version from build directory will be used rather than a
system one

### CODE REVIEW

File: sycl/test/lit.cfg.py [hunk 1]
```diff
@@ -198,10 +198,8 @@ def getDeviceCount(device_type):
 aot_tools = ["opencl-aot", "ocloc", "aoc"]
 for aot_tool in aot_tools:
     if find_executable(aot_tool) is not None:
-        print("Found AOT device compiler " + aot_tool)
+        lit_config.note("Found AOT device compiler " + aot_tool)
         config.available_features.add(aot_tool)
-    else:
-        print("Could not find AOT device compiler " + aot_tool)
```
Comment (Fznamznon) [inline sycl/test/lit.cfg.py:204]: Why did you remove this?

Comment (AlexeySachkov) [reply sycl/test/lit.cfg.py:204]: I'm not sure that this one is very valuable: for example, we don't print "Not found available CPU/GPU/accelerator device". But if you want, I can return this message back

Comment (Fznamznon) [reply sycl/test/lit.cfg.py:204]: I thought we print `getDeviceCount cpu:OpenCL runtime not found·` because it was in PR description. But in my environment I see `getDeviceCount gpu:PI_OPENCL` when I don't have GPU, this is pretty useless... I think we should print that we didn't find something too (including device runtimes and AOT tools), It could help to prevent hours of debugging nonexistent/missed problems.

Comment (AlexeySachkov) [reply sycl/test/lit.cfg.py:204]: > I thought we print getDeviceCount cpu:OpenCL runtime not found· because it was in PR description. 

Oh, I clearly missed that one :)

I will return "aot device compiler not found" message and revise other messages to double-check that they are actually useful

Comment (AlexeySachkov) [reply sycl/test/lit.cfg.py]: >  But in my environment I see getDeviceCount gpu:PI_OPENCL when I don't have GPU, this is pretty useless...

Should be fixed now

> . I think we should print that we didn't find something too (including device runtimes and AOT tools), It could help to prevent hours of debugging nonexistent/missed problems.

Done

File: sycl/test/lit.cfg.py [hunk 2]
```diff
@@ -26,49 +26,36 @@
...
+config.excludes = ['Inputs']
 
 # test_source_root: The root path where tests are located.
 config.test_source_root = os.path.dirname(__file__)
 
 # test_exec_root: The root path where tests should be run.
 config.test_exec_root = os.path.join(config.sycl_obj_root, 'test')
 
+# Propagate some variables from the host environment.
+llvm_config.with_system_environment(['PATH', 'OCL_ICD_FILENAME', 'SYCL_DEVICE_ALLOWLIST'])
```
Comment (alexbatashev) [inline sycl/test/lit.cfg.py:38]: I'd also include config path `SYCL_CONFIG_FILE_NAME`

File: sycl/test/lit.cfg.py [hunk 3]
```diff
@@ -26,49 +26,36 @@
...
-        config.environment['LIB'] = os.path.pathsep.join((config.environment['LIB'], config.sycl_libs_dir))
-    else:
-        config.environment['LIB'] = config.sycl_libs_dir
-
-    if 'PATH' in os.environ:
-        config.environment['PATH'] = os.path.pathsep.join((config.environment['PATH'], config.sycl_tools_dir))
-    else:
-        config.environment['PATH'] = config.sycl_tools_dir
+    llvm_config.with_system_environment('LIB')
+    llvm_config_with_environment('LIB', config.sycl_libs_dir, append_path=True)
```
Comment (AlexeySachkov) [inline sycl/test/lit.cfg.py:49]:
```suggestion
    llvm_config.with_environment('LIB', config.sycl_libs_dir, append_path=True)
```

File: sycl/test/lit.cfg.py (hunk 2 above)
Comment (AlexeySachkov) [reply sycl/test/lit.cfg.py]: Done

File: sycl/test/lit.cfg.py [hunk 4]
```diff
@@ -161,47 +150,52 @@ def getDeviceCount(device_type):
...
+else:
+    lit_config.warning("Accelerator device not found")
 config.substitutions.append( ('%ACC_RUN_PLACEHOLDER',  acc_run_substitute) )
 config.substitutions.append( ('%ACC_CHECK_PLACEHOLDER',  acc_check_substitute) )
 
 # PI API either supports OpenCL or CUDA.
-opencl = False
 if not cuda:
-    opencl = True
     config.available_features.add('opencl')
```
Comment (bader) [inline sycl/test/lit.cfg.py:175]: I think this code is not quite right. `opencl` feature should be enabled only if there is at least on OpenCL device in the system.
Could we fix this issue within this refactoring?

File: sycl/test/lit.cfg.py [hunk 5]
```diff
@@ -161,47 +150,52 @@ def getDeviceCount(device_type):
         gpu_check_on_linux_substitute = "| FileCheck %s"
         if cuda:
             gpu_run_on_linux_substitute += " SYCL_BE=PI_CUDA "
+else:
+    lit_config.warning("GPU device not found")
```
Comment (Fznamznon) [inline sycl/test/lit.cfg.py:154]: Not having GPU is a pretty common situation. Should it be a note instead?

File: sycl/test/lit.cfg.py [hunk 6]
```diff
@@ -161,47 +150,52 @@ def getDeviceCount(device_type):
...
 config.substitutions.append( ('%ACC_CHECK_PLACEHOLDER',  acc_check_substitute) )
 
 # PI API either supports OpenCL or CUDA.
-opencl = False
 if not cuda:
-    opencl = True
     config.available_features.add('opencl')
 
+if cuda:
+    config.substitutions.append( ('%sycl_triple',  "nvptx64-nvidia-cuda-sycldevice" ) )
```
Comment (Fznamznon) [inline sycl/test/lit.cfg.py:178]: What if I have both GPU devices CUDA and OpenCL. Will I be able to run tests on both devices?

File: sycl/test/lit.cfg.py (hunk 5 above)
Comment (AlexeySachkov) [reply sycl/test/lit.cfg.py:154]: I would like this to be a warning - it clearly highlights that the coverage of testing is reduced (in terms of devices). Just to avoid the situations when someone wanted to fix something on GPU, but forgot to properly configure environment - all tests passed, but not on GPU

File: sycl/test/lit.cfg.py (hunk 6 above)
Comment (AlexeySachkov) [reply sycl/test/lit.cfg.py:178]: If I understand correctly, yes, but you need two separate launches of the llvm-lit with different value of  `SYCL_BE` argument

File: sycl/test/lit.cfg.py (hunk 4 above)
Comment (AlexeySachkov) [reply sycl/test/lit.cfg.py:182]: Should be fixed now
